src/tests/conformance/post_handshake/validators.rs

```diff
@@ -55,3 +85,212 @@ async fn c015_TM_VALIDATOR_LIST_COLLECTION_node_should_send_validator_list() {
+    let master_public_hex = "02ED521B8124454DD5B7769C813BD40E8D36E134DD51ACED873B49E165327F6DF2";
+    let master_secret_bytes = hex::decode(master_secret_hex).expect("unable to decode hex");
+    let master_public_bytes = hex::decode(master_public_hex).expect("unable to decode hex");
+    let master_secret_key =
+        SecretKey::from_slice(master_secret_bytes.as_slice()).expect("unable to create secret key");
+
+    let signing_secret_hex = "00F963180681C0D1D51D1128096B8FF8668AFDC41CBDED511D12D390105EFDDC";
+    let signing_public_hex = "03859B76317C8AA64F2D253D3547831E413F2663AE2568F7A17E85B283CC8861E4";
+    let signing_secret_bytes = hex::decode(signing_secret_hex).expect("unable to decode hex");
+    let signing_public_bytes = hex::decode(signing_public_hex).expect("unable to decode hex");
+    let signing_secret_key = SecretKey::from_slice(signing_secret_bytes.as_slice())
+        .expect("unable to create secret key");
+
+    // 2. Create signable manifest with sequence, public key, signing public key (without signatures)
+    assert_eq!(
+        master_public_bytes.len(),
+        PUBLIC_KEY_SIZE,
+        "invalid master public key length: {}",
+        master_public_bytes.len()
+    );
+    assert_eq!(
+        signing_public_bytes.len(),
+        PUBLIC_KEY_SIZE,
+        "invalid signing public key length: {}",
+        master_public_bytes.len()
+    );
+    let signable_manifest =
+        create_signable_manifest(1, &master_public_bytes, &signing_public_bytes);
+
+    // 3. append manifest prefix
+    let mut prefixed_signable = BytesMut::with_capacity(1024);
+    prefixed_signable.put(&b"MAN\x00"[..]);
+    prefixed_signable.extend_from_slice(&signable_manifest);
+
+    // 4. Sign the signable manifest with master secret key, get master signature
+    let master_signature_bytes = sign_buffer(&master_secret_key, &prefixed_signable);
+
+    // 5. Sign it with signing private key, get signature
+    let signature_bytes = sign_buffer(&signing_secret_key, &prefixed_signable);
+
+    // 6. Create final manifest with sequence, public key, signing public key, master signature, signature
+    let manifest = create_final_manifest(
+        1,
+        &master_public_bytes,
+        &signing_public_bytes,
+        &master_signature_bytes,
+        &signature_bytes,
+    );
+
+    // 7. Create Validator blob.
+    let validator_blob = create_validator_blob_json(&manifest, master_public_hex);
+    let bstr = base64::encode(&validator_blob);
+    let blob_bytes = base64::decode(&bstr).expect("unable to decode a blob");
+    let bb = bstr.as_bytes().to_vec();
+
+    // 8.  Get signature for blob using master private key
+    let blob_signature_bytes = sign_buffer(&signing_secret_key, &blob_bytes);
+
+    // 9. Setup payload, send it
+    let mstr = base64::encode(manifest);
+    let mb = mstr.as_bytes().to_vec();
+    let sstr = hex::encode_upper(blob_signature_bytes);
+    let sb = sstr.as_bytes().to_vec();
```

Comment:
Small nit/suggestion: perhaps we could name these variables more explicitly, a few chars less isn't necessarily worth the loss in "understandable at a glance" quality?